bootstrap.py

```diff
@@ -149,15 +156,46 @@ def check_migration_version():
         sys.exit(1)
 
 
+def setup_yum_repo(url, gpg_key):
+    """
+    Configures a yum repository at /etc/yum.repos.d/katello-client-bootstrap-deps.repo
+    """
+    submanrepoconfig = SafeConfigParser()
+    submanrepoconfig.add_section('kt-bootstrap')
+    submanrepoconfig.set('kt-bootstrap', 'name', 'Subscription-manager dependencies for katello-client-bootstrap')
+    submanrepoconfig.set('kt-bootstrap', 'baseurl', url)
+    submanrepoconfig.set('kt-bootstrap', 'enabled', '1')
+    submanrepoconfig.set('kt-bootstrap', 'gpgcheck', '1')
+    submanrepoconfig.set('kt-bootstrap', 'gpgkey', gpg_key)
+    submanrepoconfig.write(open('/etc/yum.repos.d/katello-client-bootstrap-deps.repo', 'w'))
+    print_generic('Building yum metadata cache. This may take a few minutes')
+    call_yum('makecache')
+
+
 def install_prereqs():
     """
     Install subscription manager and its prerequisites.
     """
-    print_generic("Installing subscription manager prerequisites")
-    yum("remove", "subscription-manager-gnome")
-    yum("install", "subscription-manager 'subscription-manager-migration-*'")
+    print_generic("Checking subscription manager prerequisites")
+    if options.deps_repository_url:
+        print_generic("Enabling %s as a repository for dependency RPMs" % options.deps_repository_url)
+        setup_yum_repo(options.deps_repository_url, options.deps_repository_gpg_key)
+    yb = yum.YumBase()
+    pkg_list = yb.doPackageLists(patterns=['subscription-manager'])
+    call_yum("remove", "subscription-manager-gnome")
+    if pkg_list.installed:
+        print_generic("subscription-manager is installed already. Attempting update")
+        call_yum("update", "subscription-manager 'subscription-manager-migration-*'", True)
```

Comment:
in the case sub-man is installed, but sub-man-migration is not, this will not install sub-man-migration :(